src/main/java/docSharing/controller/UserController.java

```diff
@@ -35,43 +37,70 @@
 @RequestMapping("/user")
 public class UserController {
 
+    private static Logger logger = LogManager.getLogger(UserController.class.getName());
+
     @Autowired
     private UserService userService;
     @Autowired
     private DocumentService documentService;
     @Autowired
     private EmailService emailService;
 
-
+    /**
+     * deleteUserById not used
+     * @param id - users id
+     * @return -
+     */
     @RequestMapping(value = "/delete/{id}")
     public ResponseEntity<?> deleteUserById(@PathVariable("id") int id) {
         return ResponseEntity.noContent().build();
     }
 
+    /**
+     * givePermission is a PATCH function for changing the user role.
+     * @param documentId - the document id in database
+     * @param uid - the user id in database that will change his permission
+     * @param permission - the new permission
+     * @param userId - the user id that sent this request
+     * @return ResponseEntity with the message, if it worked or not.
+     */
     @RequestMapping(value = "/permission/give", method = RequestMethod.PATCH)
     public ResponseEntity<?> givePermission(@RequestParam Long documentId, @RequestParam Long uid, @RequestParam Permission permission, @RequestAttribute Long userId) {
+        logger.info("in UserController -> givePermission");
         if (documentId == null || uid == null || permission == null) {
+            logger.error("in UserController -> givePermission -> on of documentId,uid,permission is null");
             return ResponseEntity.badRequest().build();
         }
-
         try {
             // FIXME: should be in the filter -> permission filter
             if (!Objects.equals(documentService.findById(documentId).getUser().getId(), userId)) {
+                logger.warn("in UserController -> givePermission -> " +ExceptionMessage.USER_IS_NOT_THE_ADMIN);
```

Comment:
This check doesn't have to be here at all. Read the comment on line 75.
You need to move it to the permission filter and check if the user who wants to manipulate the file or whatever, he has to be the owner of it.